packages/ai/src/agent.ts

```diff
@@ -464,6 +465,42 @@ export class AIAgent extends BaseActor<AgentMetadata> implements IIAgent {
     );
   }
 
+  private async _withInlineMcps(mcps?: string[]): Promise<ToolsetsInput> {
```

Comment:
I think you should change `_getOrCreateCallableToolSet` instead and receive string | MCPConnection

then you need to change `_pickCallableTools` to use metadata?.toolsets
[...Object.entries(tool_set), ...(this.metadata?.toolsets??[]).map({connection, filters} => [connection, filters])]

Reply:
toolsets: [{connection: MCPConnection, filters: []}]